website/src/pages/tech-specs.md

```diff
@@ -0,0 +1,350 @@
+| ------------- | ------------------------------------------------------------ |
+| Copy on Write | Optimized for read performance and ideal for slow changing datasets |
+| Merge-on-read | Optimized to balance the write and read performance and ideal for frequently changing datasets |
+
+
+
+### Data Model
+
+Within each partition, data is organized into key-value model. Every row is uniquely identified with a row key. To write a row into Hudi dataset, each row must specify the following user fields 
+
+| User fields                 | Description                                                  |
+| --------------------------- | ------------------------------------------------------------ |
+| Partitioning key [Optional] | Value of this field defines the directory hierarchy within the table base path. This essentially provides an hierarchy isolation for managing data and related metadata |
+| Row key                     | Record keys uniquely identify a record/row within each partition if partitioning is enabled |
+| Ordering field(s)           | Hudi guarantees the uniqueness constraint of row key and the conflict resolution configuration manages strategies on how to disambiguate when multiple records with the same keys are to be merged into the dataset. The resolution logic can be based on an ordering field or can be custom, specific to the dataset. To ensure consistent behavior dealing with duplicate records, the resolution logic should be commutative and idempotent |
+
+**Hudi metadata fields**
+
+Hudi format stores the user fields along with the row merged along with transactional metadata fields. These fields are encoded in the data-file format and available in the table schema. 
+
+[TODO - flush this section out with details on guarantees and how to populate them]
+
+| Husi meta-fields              | Description                                                  |
+| ----------------------------- | ------------------------------------------------------------ |
+| _hoodie_commit_time [string]  | Every modification to a Hudi dataset creates an entry into the Transaction timeline. This entry is identified with the commit time. This field matches to the commit time of the instant in the timeline that created this record. More on this in Hudi transactions section below. |
+| _hoodie_commit_seqno [string] | A unique sequencing number identifying the position of the record within each commit [TODO - where do we use this?] |
+| _hoodie_record_key            | Unique record key identifying the row within the partition   |
+| _hoodie_partition_path        | Partition path under which the data is organized into        |
+| _hoodie_file_name             | The data file name this record belongs to                    |
+| _hoodie_is_deleted            | Tombstone field to denote the record key is deleted          |
+| _hoodie_operation             | [TODO] Is this a CDC Field? Not sure where this is used.     |
```

Comment:
Yes, this is for CDC support in Flink. If changelog is enabled in Flink, then this field could take one of the values in `HoodieOperation` and it is used in merging log files while reading hudi table in flink (`MergeOnReadInputFormat`). @danny0405 can add more details here.